popup.js

```diff
@@ -1,247 +1,109 @@
-			document.body.appendChild(setSecretBtn);   
-document.body.appendChild(document.createElement("br"));
-document.body.appendChild(document.createElement("br"));
-
-			getSecretBtn.addEventListener('click', readSecret , false);
-			setSecretBtn.addEventListener('click', writeSecret , false);
-	    },
-	    error: function(data) {
-	        notify.textContent = "ERROR: " + JSON.stringify( data ) 
-	    }
-	});
+  })
+  chrome.storage.sync.get(['vaultAddress'], function (result) {
+    vaultServerAdress = result.vaultAddress
+  })
+
+  chrome.storage.sync.get(['secrets'], function (result) {
+    secretList = result.secrets
+    if (!secretList) {
+      secretList = []
+    }
+    resultList.textContent = ''
+
+    secretList.forEach(secret => {
+      jQuery.ajax({
+        type: 'LIST',
+        url: vaultServerAdress + '/v1/secret/metadata/vaultPass/' + secret,
+        headers: { 'X-Vault-Token': vaultToken },
+        contentType: 'application/json',
+        dataType: 'json',
+        success: function (data) {
+          data.data.keys.forEach(element => {
+            var pattern = new RegExp(element)
+            var patternMatches = pattern.test(currentUrl)
+            console.log(element + ' - ' + currentUrl + ' - ' + patternMatches)
+            if (patternMatches) {
+              getCredentials(vaultServerAdress + '/v1/secret/data/vaultPass/' + secret + element).then((credentials) =>
+                addCredentials(credentials.data.data, element, resultList))
+            }
+          })
+        },
+        error: function (data) {
+          console.error('ERROR accessing ' + secret + ': ' + JSON.stringify(data))
+        }
+      })
+    })
+  })
 }
 
-// invoked after user clicks "login to vault" button, if all fields filled in, and URL passed regexp check.
-function authToVault( myurl, username, password ) {
-
-notify.textContent = "server: " + myurl + "/v1/auth/userpass/login/" + username;
-sleep(500)
-
-	jQuery.ajax({
-	    type: "POST",
-	    url: myurl + "/v1/auth/userpass/login/" + username,
-	    data: JSON.stringify( {password: password} ),
-	    contents: "json",
-	    dataType: "json",
-	    success: function(data) {
-	        notify.textContent = "AUTH SUCCESS, token: " + data.auth.client_token ;
-	        mounts.textContent = "mounts list here"
-sleep(500)
-	        // set global token for other functions
-	        userToken = data.auth.client_token
-
-	        //clear page from auth items
-	        clearPage()
-
-	        // pass those params to showMounts function
-	        var params = {
-	        	token: data.auth.client_token,
-	        	url: myurl ,
-	        	urlSuffix: "/v1/sys/mounts"
-	        }
-	        //show all mounts
-	        showMounts( params )
-	    },
-	    error: function(data) {
-	        notify.textContent = "ERROR: " + JSON.stringify( data ) ;
-	    }
-	});
-
+function addCredentials(credentials, credentialName, list) {
+  var item = document.createElement('li')
+  console.log(credentials)

```

Comment:
This doesn't feel right :]

Reply:
you mean the log of credentials to console? yeah.. I would remove that too I guess.

Reply:
I'll quickly review all the console calls and will remove them

Reply:
https://github.com/zeichenanonym/vault-chrome-extension/commit/c5360d8aa59ea7e17f5da977d12ef8daa2984b7c resolves this